internal/hcsoci/resources_lcow.go

```diff
@@ -35,7 +35,7 @@ func getGPUVHDPath(coi *createOptionsInternal) (string, error) {
 	return gpuVHDPath, nil
 }
 
-func allocateLinuxResources(ctx context.Context, coi *createOptionsInternal, r *resources.Resources) error {
+func allocateLinuxResources(ctx context.Context, coi *createOptionsInternal, r *resources.Resources, sandbox bool) error {
```

Comment:
I don't know how I feel about passing around `sandbox` when we still have access to the entire `coi` here to parse and determine the container type. 

If we keep this though, I think we should rename `sandbox` to `isSandbox`

Reply:
I didn't want to reparse the container type as we already do that earlier in `CreateContainer`. I'm fine to meet in the middle with isSandbox

Reply:
I'm totally fine with `isSandbox`. Though I don't think reparsing should really be a concern here since it's essentially an O(1) call.

Reply:
I'm not concerned about perf, just duplicated code